packages/nest/README.md

```diff
@@ -0,0 +1,125 @@
+<!-- markdownlint-disable MD033 -->
+<!-- x-hide-in-docs-start -->
+<p align="center">
+  <picture>
+    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/white/openfeature-horizontal-white.svg" />
+    <img align="center" alt="OpenFeature Logo" src="https://raw.githubusercontent.com/open-feature/community/0e23508c163a6a1ac8c0ced3e4bd78faafe627c7/assets/logo/horizontal/black/openfeature-horizontal-black.svg" />
+  </picture>
+</p>
+
+<h2 align="center">OpenFeature Nest.js SDK</h2>
+
+<!-- x-hide-in-docs-end -->
+<!-- The 'github-badges' class is used in the docs -->
+<p align="center" class="github-badges">
+  <a href="https://github.com/open-feature/spec/releases/tag/v0.7.0">
+    <img alt="Specification" src="https://img.shields.io/static/v1?label=specification&message=v0.7.0&color=yellow&style=for-the-badge" />
+  </a>
+  <br/>
+  <a href="https://codecov.io/gh/open-feature/js-sdk">
+    <img alt="codecov" src="https://codecov.io/gh/open-feature/js-sdk/branch/main/graph/badge.svg?token=3DC5XOEHMY" />
+  </a>
+</p>
+<!-- x-hide-in-docs-start -->
+
+[OpenFeature](https://openfeature.dev) is an open specification that provides a vendor-agnostic, community-driven API
+for feature flagging that works with your favorite feature flag management tool.
+
+🧪 This is SDK is experimental.
+
+#### Here's a basic example of how ot use the current API with flagd.
+
+#### Registering the Nest.js SDK module in the App Module:
+
+```ts
+import { Module } from '@nestjs/common';
+import { FlagdProvider } from '@openfeature/flagd-provider';
+import { OpenFeatureModule } from '@openfeature/nestjs-sdk';
+
+@Module({
+  imports: [
+    OpenFeatureModule.forRoot({
+      defaultProvider: new FlagdProvider({
+        host: 'localhost',
+        port: 8013,
+        tls: false,
+      }),
+      providers: {
+        differentServer: new FlagdProvider({
+          host: 'different-server.com',
+          port: 8013,
+          tls: false,
+        }),
+      },
+    }),
+  ],
+})
+export class AppModule {
+}
+```
+
+#### Injecting a feature flag with header value in evakuation context into an endpoint handler method
+
+```ts
+import { Controller, ExecutionContext, Get } from '@nestjs/common';
+import { map, Observable } from 'rxjs';
+import { BooleanFeatureFlag } from '@openfeature/nestjs-sdk';
+import { EvaluationDetails } from '@openfeature/server-sdk';
+import { Request } from 'express';
+
+function getContext(executionContext: ExecutionContext) {
```

Comment:
Got it!

One concern here is how it can be propagated to different layers (modules/services) since.
`ExecutionContext` can only be obtained through  at controller level, right?

Maybe using [async local storage](https://docs.nestjs.com/recipes/async-local-storage) with [interceptors](https://docs.nestjs.com/interceptors) this can be a bit more ergonomic and accessible within different layers. 

Getting the type of execution context we could check for specific metadata fields (in http case headers) to check and configure an async local storage

Reply:
My feeling is that the FeatureFlag decorators only make sense for the endpoint handlers.
In the services, it feels like one would use the injected clients. To get request information I would assume one would use request scopes services that build the context themselves.

> Maybe using [async local storage](https://docs.nestjs.com/recipes/async-local-storage) with [interceptors](https://docs.nestjs.com/interceptors) this can be a bit more ergonomic and accessible within different layers.

You could implement a PR for this or maybe try it out. I am not sure how I would do this or if this could be made with good ergonomics.

Reply:
> Maybe using async local storage with interceptors this can be a bit more ergonomic and accessible within different layers.

I really like the decorator approach, and I think it still would be cool to have them in lower-level modules. I think this might be possible in all layers if you used nest [factory dependency](https://docs.nestjs.com/fundamentals/custom-providers#factory-providers-usefactory) which provided the context, is injected in `Scope.REQUEST`, and itself received the `req` dependency. In my experience with NestJS, whenever I wanted request data lower down the stack, I'd just add `Scope.REQUEST` dependencies to the lower modules and use it there to avoid passing whatever data I need down the entire class-stack manually. As long as the objects you're constructing can be created per-request with low memory cost it works well.

I'm not exactly sure how much of these we'd want to implement, and how much we'd just give examples for and leave to the consumer.

Reply:
I would propose that we make a new issue for this, that we can work on then :)